includes/class-wc-stripe-webhook-handler.php

```
@@ -1385,6 +1385,80 @@ public function process_deferred_webhook( $webhook_type, $additional_data, $noti
+					} else {
+						// Still no order found after deferral — handle as an agentic checkout session.
+						WC_Stripe_Logger::debug( "No order found for deferred {$webhook_type} ($session_id). Handling as agentic session." );
+						$this->handle_agentic_checkout_session( $notification );
+					}
+					break;
+				case 'checkout.session.expired':
+				case 'checkout.session.async_payment_failed':
+					$order = isset( $additional_data['order_id'] ) ? wc_get_order( $additional_data['order_id'] ) : null;
+
+					if ( empty( $order ) ) {
+						throw new Exception( "Missing required data. 'order_id' is invalid or not found for the deferred '{$webhook_type}' event." );
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '1427', column '17'). (undefined)

(MissingImport)